src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexRunner.cs

```diff
@@ -65,85 +68,42 @@ public static RegexRunnerFactory CreateFactory(RegexNode rootNode, RegexOptions
                         (options & RegexOptions.RightToLeft) != 0 ? nameof(RegexOptions.RightToLeft) : nameof(RegexOptions.ECMAScript)));
             }
 
-            return new SymbolicRegexRunnerFactory(
-                new SymbolicRegexRunner(rootNode, options, matchTimeout, culture));
+            return new SymbolicRegexRunnerFactory(new SymbolicRegexRunner(code, options, matchTimeout, culture));
         }
+        protected override void InitTrackCount() { } // nop, no backtracking
 
-        internal override Match? ScanInternal(
-            Regex regex, string text, int textbeg, int textend, int textstart, int prevlen, bool quick, TimeSpan timeout)
-        {
-            int length = textend - textbeg;
-
-            // If the previous match was empty, advance by one before matching
-            // or terminate the matching if there is no remaining input to search in
-            if (prevlen == 0)
-            {
-                if (textstart == textend)
-                {
-                    return Match.Empty;
-                }
-
-                textstart += 1;
-            }
+        protected override bool FindFirstChar() =>
+            // The real logic is all in Go.  Here we simply validate if there's enough text remaining to possibly match.
+            runtextpos <= runtextend - _minRequiredLength;
 
-            SymbolicMatch pos = _matcher.FindMatch(quick, text, textstart, textend);
-            if (!pos.Success)
-            {
-                return Match.Empty;
-            }
-
-            if (quick)
+        protected override void Go()
+        {
+            // TODO: In order to produce a SymbolicMatch with a valid Index/Length, FindMatch needs to do additional
+            // work once it's found the end of a match to find the beginning of the match.  If this is an IsMatch
+            // call, however, the details of the match will be ignored, with the only important information being
+            // that there was a match.  We need a mechanism of supplying that information to this operation. At the
+            // moment the only obvious answer to that is new protected surface area, either additional protected
+            // field/property that could be consulted, or a new virtual Scan method that this implementation could
+            // override instead of using the base Scan method that invokes FindFirstChar/Go.  We could choose to bypass
+            // this public/protected surface area for the in-memory implementation, e.g. having it look at some
+            // new private protected field we add to RegexRunner, but for now we simply don't benefit from this optimization.
+            const bool calledFromIsMatch = false;
```

Comment:
There are some cases where the perf difference can be substantial. 
Example regex: `.{20}a[a-z]*$` , `input = "01234567890123456789xsomelongrandomtext...goingonandon"`
The forward pass will be very easy to test that a match exists (a linear pass). But finding where the match starts will involve using the reverse of this regex that is  `^[a-z]*a.{20}` (walking the input in reverse) that is the classical regex causing bowup in the reverse DFA (that may then switch to Antimirov mode) in order to find the start of the match, that in the end is actually irrelevant.